### PR TITLE
Include additional formats in the datepicker

### PR DESCRIPTION
**Pull request by oGalceran1**
Include additional formats required by Anthema team as well as the format d/mm/yyyy or dd/m/yyyy or d/mm/yy or dd/m/yy.
Formats are i18n tested

### CODE REVIEW

File: src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts [hunk 1]
```diff
@@ -23,12 +23,16 @@ export class USMockI18nService {
 	}
 
 	public getDateFormatForDatePicker() {
-		return 'mm/dd/yy';
+		return 'MM/DD/YY';
```
Comment (XavierCalvo) [inline src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts]: This mock is not returning the same that original service

- systelab-translate/i18n.service.ts/getDateFormatForDatePicker   
-->  calls to --> systelab-translate/ date-util.ts/getDateFormatForDatePicker   

https://github.com/systelab/systelab-translate/blob/master/src/app/systelab-translate/date-util/date-util.ts

```
	public getDateFormatForDatePicker(isFullYear = false): string {
		let stringDateFormat = '';
		switch (this.locale) {
			case 'en-US':
				stringDateFormat = 'm/d/y';
				break;
...
		if (isFullYear) {
			stringDateFormat = stringDateFormat.replace('y', 'yy');
			if (this.locale === 'en-US') {
				stringDateFormat = stringDateFormat.replace('m', 'mm');
				stringDateFormat = stringDateFormat.replace('d', 'dd');
			}
		}
		return stringDateFormat;
```

File: src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts [hunk 2]
```diff
@@ -41,12 +45,16 @@ export class ESMockI18nService {
 	}
 
 	public getDateFormatForDatePicker() {
-		return 'dd/mm/yy';
+		return 'DD/MM/YY';
```
Comment (XavierCalvo) [inline src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts]: Idem than USMock comment

File: src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts (hunk 1 above)
Comment (sgsergiogracia) [reply src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts]: You're right!

File: src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts (hunk 2 above)
Comment (oGalceran1) [reply src/app/systelab-components/datepicker/datepicker.component.i18n.spec.ts]: All changes have been applied

File: src/app/systelab-components/datepicker/date-transformer.service.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+	private getDateSeparator(dateFormat: string) {
+		let dateSeparator: string;
+		if (dateFormat.lastIndexOf('/') > 0) {
+			dateSeparator = '/';
+		} else if (dateFormat.lastIndexOf('-') > 0) {
+			dateSeparator = '-';
+		} else if (dateFormat.lastIndexOf('.') > 0) {
+			dateSeparator = '.';
+		}
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/date-transformer.service.ts]: Be coherent with the following method and use includes.

File: src/app/systelab-components/datepicker/date-transformer.service.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+		return dateSeparator;
+	}
+
+	private hasSeparator(dateTmp: string): boolean {
+		return dateTmp.includes('/') || dateTmp.includes('-') || dateTmp.includes('.');
+	}
+
+	private removeSeparator(dateTmp: string, dateSeparator: string): string {
+		return dateTmp.split(dateSeparator)
+			.join('');
```
Comment (alfonsserra) [inline src/app/systelab-components/datepicker/date-transformer.service.ts]: Is there any chance to use a regular expression?

**Pull request merged**